### PR TITLE
Fixes some compiler warnings

### PR DESCRIPTION
**Pull request by AlexChristoforides**

### CODE REVIEW

File: htword/WordKeyInfo.cc [hunk 1]
```diff
@@ -45,7 +45,7 @@ int WordKeyField::SetNum(WordKeyField *previous, char *nname, int nbits)
 
 int WordKeyField::SetString()
 {
-  name.set("Word");
+  strcpy(name, "Word");
```
Comment (andy5995) [inline htword/WordKeyInfo.cc]: Please see http://www.tutorialspoint.com/cprogramming/c_structures.htm about variables with periods in them.

Comment (AlexChristoforides) [reply htword/WordKeyInfo.cc]: Sorry! I didn't realize it was a struct!

File: htword/WordKey.cc [hunk 2]
```diff
@@ -369,8 +369,9 @@ WordKey::Prefix() const
     //
     // Fields set, then fields unset then field set -> not a prefix
     //
-    if(IsDefined(j))
+    if(IsDefined(j)) {
```
Comment (andy5995) [inline htword/WordKey.cc]: We're going with the GNU style formatting (see the note in #documentation), which means the brace  should be moved to the next line after the if statement.

https://www.gnu.org/prep/standards/html_node/Formatting.html

I know the rest of the code doesn't match yet. Kind of a pain, but keeping everything uniform makes things easier and more efficient for collaborators in the long-term.

File: htlib/StringMatch.cc [hunk 3]
```diff
@@ -311,7 +311,7 @@ int StringMatch::Compare(const char *string, int &which, int &length)
     //
     while ((unsigned char)string[pos])
     {
-  new_state = table[trans[string[pos]]][state];
+  new_state = table[trans[(unsigned int)string[pos]]][state];
```
Comment (andy5995) [inline htlib/StringMatch.cc]: So once you refactor this and take care of the other changes, I can merge this. Done commenting. :)

**Pull request merged**